src/main/java/uk/gov/hmcts/reform/iacasepaymentsapi/domain/handlers/presubmit/PaymentAppealPreparer.java

```diff
@@ -77,28 +82,38 @@ public boolean canHandle(
                 .getCaseDetails()
                 .getCaseData();
 
+        PreSubmitCallbackResponse<AsylumCase> response =
+            new PreSubmitCallbackResponse<>(callback.getCaseDetails().getCaseData());
+
         Optional<RemissionType> optRemissionType = asylumCase.read(REMISSION_TYPE, RemissionType.class);
         if ((optRemissionType.isPresent() && optRemissionType.get() == RemissionType.NO_REMISSION)
             || optRemissionType.isEmpty()) {
 
-            List<String> accountsFromOrg = refDataService.getOrganisationResponse()
-                .getOrganisationEntityResponse().getPaymentAccount();
+            try {
 
-            if (accountsFromOrg.isEmpty()) {
-                PreSubmitCallbackResponse<AsylumCase> response =
-                    new PreSubmitCallbackResponse<>(callback.getCaseDetails().getCaseData());
-                response.addErrors(Collections.singleton("There are no payment accounts"));
+                List<String> accountsFromOrg = refDataService.getOrganisationResponse()
+                    .getOrganisationEntityResponse().getPaymentAccount();
 
-                return response;
-            }
+                if (accountsFromOrg.isEmpty()) {
+                    response.addErrors(Collections.singleton("There are no payment accounts"));
+
```

Comment:
asylumCase.write(HAS_PBA_ACCOUNTS, YesOrNo.NO);
Will it be useful to have this set here ?